File: ASProj/app/src/main/java/org/devio/as/proj/main/MainActivity.java
package org.devio.as.proj.main;

import android.os.Bundle;

import androidx.annotation.NonNull;

import org.devio.as.proj.common.ui.component.HiBaseActivity;
import org.devio.as.proj.main.logic.MainActivityLogic;
import org.jetbrains.annotations.NotNull;

public class MainActivity extends HiBaseActivity implements MainActivityLogic.ActivityProvider{
    private MainActivityLogic activityLogic;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        activityLogic = new MainActivityLogic(this, savedInstanceState);
    }

    @Override
    protected void onSaveInstanceState(@NonNull @NotNull Bundle outState) {
        super.onSaveInstanceState(outState);
        activityLogic.onSaveInstanceState(outState);
    }
}

File: hi-library/app/src/main/java/org/devio/hi/library/app/MainActivity.kt
package org.devio.hi.library.app

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import androidx.lifecycle.lifecycleScope
import kotlinx.coroutines.launch
import org.devio.hi.library.coroutine.CoroutineScene3
import org.devio.hi.library.coroutine.CountDownLatchDemo
import org.devio.hi.library.coroutine.SemaphoreDemo

class MainActivity : AppCompatActivity(), View.OnClickListener {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onClick(v: View) {
        when(v.id) {
            R.id.tv_hilog -> {
                SemaphoreDemo.main()
            }
        }
    }
}
